app/src/main/res/values/strings.xml

```diff
@@ -85,9 +85,9 @@
     <string name="about_title_license">"License"</string>
     <string name="about_summary_license">"GNU General Public License"</string>
     <string name="offline">"You are offline"</string>
-    <string name="confirmation_authorize_now">"You need to authorize with your OSM user account to apply any of your answers for good. Authorize now?"</string>
+    <string name="confirmation_authorize_now">"You need to authorize with your OSM user account to upload your answers. Authorize now?"</string>
```

Comment:
The wording is the result of an attempt to avoid semi-technical terms like "upload" and "download" because by default, the user does not need to do any manual up- or download, everything is applied automatically. I find "apply" better.

Reply:
However, let's don't do it too technical. As a dev you may always try to describe everything technically 100% correct and see it through the "dev's eyes", but users actually won't care and rather prefer a simple, understandable language. Don't forget these are two languages we are talk... 😉 

Users should just understand here, that they need to login for their answers to have any effect. How you describe that the effect does not really matter (i.e. "apply"/"upload"/"contribute"/...), users should just understand the correct thing. What happens behind the scenes should not matter... 😃 

So as a compromise maybe "contribute"? As, when not logged in, the answers cannot be contributed (back to OSM).

Reply:
What I also tried to convey with the current text is that his answers will not be gone if he does not upload them now but the app will keep them until they are uploaded. This is why a wrote "apply ... for good". I am not a native speaker though.

Reply:
“Apply any of your answers for good” sounds awkward to me.  I think “upload” conveys a clearer meaning to users. 

The term upload is used on many common websites, so I don’t think it’s too technical of a term. 

![3f586248-b8e0-44f1-bfd4-283a00dd10f0](https://user-images.githubusercontent.com/304383/38104602-a7da8ace-3357-11e8-86cc-c773a92961a4.jpeg)


![3bd149f9-63fc-4cc7-8ae2-c067e56d6088](https://user-images.githubusercontent.com/304383/38104492-49842386-3357-11e8-9759-b2c665c31c37.jpeg)

If we really want to avoid using “upload”, @rugk had great input on alternatives.

Reply:
That is the point, the app is not a website. From a website, I would expect it to forget the things I entered (in a form) if I don't upload it.

Reply:
Then "save", or "submit", or "send to OSM".

Reply:
This is worse. What happens to a file you do not save and a form you do not submit?

Reply:
"Publish"?

Reply:
Here it seems for me hat new version is more confusing - especially as typically user is not supposed to upload/download anything manually.

Reply:
I think the original trigger for the change was the "for good", which I doubt is proper English. 

But apart from that, "publish" may be indeed good (or, basically the same as "send"). It does not imply the change is lost, when you do not publish it, as it is just saved offline/not published.

And BTW I do not think anyway that a user would think the change is lost. Of course the popup or where this comes, only says "Hey, login to upload the changes.". Otherwise they are just not uploaded.

As for the website comparison:
> From a website, I would expect it to forget the things I entered (in a form) if I don't upload it.

Yeah, great! So as SC is not a website, which everyone notices, the user does _not_ think what you explained, i.e. form data is not lost when not uploaded. So that term should suit your requirements?

Reply:
I am okay with "publish".

Reply:
I just changed the wording to “publish”.